src/Arcus.Security.Core/Exceptions/SecretNotFoundException.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using GuardNet;
+
+namespace Arcus.Security.Core.Exceptions
+{
+    /// <summary>
+    /// Exception, thrown when no secret was found, using the given name.
+    /// </summary>
+    [Serializable]
+    public class SecretNotFoundException : Exception
+    {
+        public SecretNotFoundException(string name) : this(name, null)
```

Comment:
Ok, last suggestion: the public docs for the ctor of this type could be valuable but this again is probably writen down in the guidelines.